### PR TITLE
Feat/update contentlayer algo

### PR DESCRIPTION
**Pull request by IgboPharaoh**
This PR adds the following:
- Updates the contentlayer algorithm for setting up the contentTree by ensuring appropriate file structure for folders and subfolders
- Updates the types algorithm to accommodate for the new types structure and for the type page
- Add the types page to the explore navigation
- Fix undefined or null types sections on the landing page
- Move sources folder and subfolder under base route

Note:
A new submodule `refine-taxonomies` which targets the changes added in [refine-taxonomies](https://github.com/bitcointranscripts/bitcointranscripts/pull/557) was added to aid testing the changes in this PR.
This will be deleted and the right branch referenced in the config before this is merged.

### CODE REVIEW

File: src/app/(explore)/[...slug]/page.tsx [hunk 1]
```diff
@@ -15,30 +15,15 @@ import Image from "next/image";
...
-    if (!lastRouteIndex) {
-      acc.push({ slug: slugAsParams });
-      // converts slug ['adopting-bitcoin', '_index'] to ['adopting-bitcoin']
-      // skips non english dir e.g ['adopting-bitcoin', '_index.es'] is skipped
-    } else if (!is_non_english_dir_or_transcript) {
-      acc.push({ slug: slugAsParams.slice(0, slugAsParams.length - 1) });
+  const allSlugs = allContentSources.map(({ slugAsParams, language }) => {
+    if (language === "en") {
+      return { slug: slugAsParams };
+    } else {
```
Comment (Emmanuel-Develops) [inline src/app/(explore)/[...slug]/page.tsx:21]: in `contentlayer.config.ts`, non english language transcripts were skipped.
Let's skip languages here as well and tackle it in a separate PR.

handling languages require some additional changes to the content tree and that is out of scope for this PR

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]

**Comment by Emmanuel-Develops** [comment]
Awesome work @IgboPharaoh 

a couple of UI nits:
- We need to display additional data when applicable at the source level. (This applies to sources and source groups). The data is already available on the allSources document.
For example; `bitcoinplusplus/layer2` source should have the metadata of website in its description
Screenshot 2024-10-27 at 3 32 37 PM
Screenshot 2024-10-27 at 11 20 23 PM

- `sources` link in the sidebar should be highlighted when on a source or sourcegroup page

File: contentlayer.config.ts [hunk 2]
```diff
@@ -224,34 +187,59 @@ function createSpeakers(transcripts: ContentTranscriptType[]) {
...
-    if (isValid) {
-      if (slugSources[slug] !== undefined) {
-        sourcesArray[slugSources[slug]].count += 1;
+    if (slugSources[slug] !== undefined) {
+      sourcesArray[slugSources[slug]].count += 1;
+    } else {
+      const sourcesLength = sourcesArray.length;
+      slugSources[slug] = sourcesLength;
+      const getTranscriptName = (slug: string) =>
+        sources.find((source) => source.language === "en" && source.slugAsParams[0] === slug)?.title ?? unsluggify(slug);
```
[the remark this answers is not in the document]
Comment (IgboPharaoh) [reply contentlayer.config.ts:206]: I have not seen an instance where we don't but this was added as a fallback for such an occurrence

File: contentlayer.config.ts [hunk 3]
```diff
@@ -224,34 +187,59 @@ function createSpeakers(transcripts: ContentTranscriptType[]) {
...
+  return { sourcesArray, slugSources };
+}
+
+const createTypesCount = (transcripts: ContentTranscriptType[], sources: ContentSourceType[]) => {
+  const { sourcesArray, slugSources } = generateSourcesCount(transcripts, sources);
+  const nestedTypes: any = {};
+
+  sources.forEach((transcript) => {
+    if (transcript.types) {
+      const type = transcript.types[0];
```
[the remark this answers is not in the document]
Comment (IgboPharaoh) [reply contentlayer.config.ts:226]: This was done because ideally a source should have only one type but this can be added

**Comment by IgboPharaoh** [comment]
> issue:
We need to display additional data when applicable at the source level. The data is already available on the allSources document.

Addition of website will be addressed as there is a section for it in the design.


Screenshot 2024-10-28 at 01 19 29

Also, I think the design clarifies what metadata to be displayed

**Comment by kouloumos** [comment]
> cc: @kouloumos, which of these fields would you consider essential, or do we display all? also, this applies to only sources and not sourcegroups correct?

@IgboPharaoh is right, the design shows the metadata to be displayed but we never clarified the mapping:

1. Primary link is `website`
2. Then secondary links are everything in `additional_resources` which is a dictionary of `title` and `url`

**Comment by Emmanuel-Develops** [comment]
A couple of other nits from dev call yesterday:
### Capitalise navigation
issue:
the navigation in `/types` should be capitalized, `call` -> `Call` etc
Screenshot 2024-10-30 at 11 00 36 AM

### Some links from wasabi research club is broken
issue: https://registry-itr05heoa-btc-knowledge-base.vercel.app/wasabi-research-club/anonymous-credentials is broken

### Breadcrumbs title
issue: Breadcrumbs should use the title of the source or transcript it is on
old registry breadcrumb, notice the title
Screenshot 2024-10-30 at 11 15 25 AM

new registry breadcrumb
Screenshot 2024-10-30 at 11 15 51 AM

cc: @IgboPharaoh

**Comment by IgboPharaoh** [comment]
> **Some links from wasabi research club is broken**
issue: https://registry-itr05heoa-btc-knowledge-base.vercel.app/wasabi-research-club/anonymous-credentials is broken

1. I investigated this issue and I noticed they were no gh-pages generated for the transcripts inside the wasabi-research-club folder under gh-pages. This is because currently this pr targets the refine-taxonomies branch and the contentTree which holds the value of the urls is generated from it. The current deployment of gh-pages is pointing to bitcoin-transcripts and there are no gh-pages generated for `wasabi-research-club` because it is non existent. What we have currently in the gh-pages is wasabi, which houses the podcast and research-club folders and it is under this folders that the respective gh-pages are present.

**Comment by Emmanuel-Develops** [comment]
> > **Some links from wasabi research club is broken**
> > issue: https://registry-itr05heoa-btc-knowledge-base.vercel.app/wasabi-research-club/anonymous-credentials is broken
> 
> 1. I investigated this issue and I noticed they were no gh-pages generated for the transcripts inside the wasabi-research-club folder under gh-pages. This is because currently this pr targets the refine-taxonomies branch and the contentTree which holds the value of the urls is generated from it. The current deployment of gh-pages is pointing to bitcoin-transcripts and there are no gh-pages generated for `wasabi-research-club` because it is non existent. What we have currently in the gh-pages is wasabi, which houses the podcast and research-club folders and it is under this folders that the respective gh-pages are present.

Good find! Yes, this seems to be the case. It is fine then since we are working off the refine-taxonomies branch that will be merged to prod.

**Comment by Emmanuel-Develops** [comment]
> > **Some links from wasabi research club is broken**
> > issue: https://registry-itr05heoa-btc-knowledge-base.vercel.app/wasabi-research-club/anonymous-credentials is broken
> 
> 1. I investigated this issue and I noticed they were no gh-pages generated for the transcripts inside the wasabi-research-club folder under gh-pages. This is because currently this pr targets the refine-taxonomies branch and the contentTree which holds the value of the urls is generated from it. The current deployment of gh-pages is pointing to bitcoin-transcripts and there are no gh-pages generated for `wasabi-research-club` because it is non existent. What we have currently in the gh-pages is wasabi, which houses the podcast and research-club folders and it is under this folders that the respective gh-pages are present.

cc: @kouloumos. I'm assuming the hugo pages build will have the correct linking when [refine-taxonomies](https://github.com/bitcointranscripts/bitcointranscripts/pull/557) branch gets merged to [master](https://github.com/bitcointranscripts/bitcointranscripts/tree/master)

@IgboPharaoh how about the other issues from the dev call, I see the breadcrumb will be challenging as we do not hold its previous context title. E.g when in a source we don't have context on it's parent (source group) title have you been able to find a way to get this data?

**Comment by IgboPharaoh** [comment]
> > > **Some links from wasabi research club is broken**
> > > issue: https://registry-itr05heoa-btc-knowledge-base.vercel.app/wasabi-research-club/anonymous-credentials is broken
> > 
> > 
> > 
> > 1. I investigated this issue and I noticed they were no gh-pages generated for the transcripts inside the wasabi-research-club folder under gh-pages. This is because currently this pr targets the refine-taxonomies branch and the contentTree which holds the value of the urls is generated from it. The current deployment of gh-pages is pointing to bitcoin-transcripts and there are no gh-pages generated for `wasabi-research-club` because it is non existent. What we have currently in the gh-pages is wasabi, which houses the podcast and research-club folders and it is under this folders that the respective gh-pages are present.
> 
> cc: @kouloumos. I'm assuming the hugo pages build will have the correct linking when [refine-taxonomies](https://github.com/bitcointranscripts/bitcointranscripts/pull/557) branch gets merged to [master](https://github.com/bitcointranscripts/bitcointranscripts/tree/master)
> 
> @IgboPharaoh how about the other issues from the dev call, I see the breadcrumb will be challenging as we do not hold its previous context title. E.g when in a source we don't have context on it's parent (source group) title have you been able to find a way to get this data?

I've explored different ways to change the way we manage the state of the bread crumb but nothing concrete yet. Currently, we're using pathname which relies on the urlparams to get the data to display in the breadcrumb but adding the exact page title will mean we're pushing state up into the breadcrumb as it is a client component which relies on the urlParams for data. A way to explore is rewriting the contentTree algorithm because it is from this structure we define what the urlParams we be like, so from the grouping of the sources we can decide to use titles as keys but I don't know how performant this will be. I'm open to ideas and solutions on the best way to go about this.

All other issue pointed out have been fixed and a commit is ready for it.  Comments have also been made to address other feedback concerning the broken links in the wasabi folder.

**Comment by Emmanuel-Develops** [comment]
> A way to explore is rewriting the contentTree algorithm because it is from this structure we define what the urlParams will be like, so from the grouping of the sources we can decide to use titles as keys but I don't know how performant this will be. I'm open to ideas and solutions on the best way to go about this.

We plan to restructure for languages, this isn't priority enough to add to this PR. Lets make the restructure when we fix languages on `source-data.json` as the source `title` can then be an individual field.

Issue added here: https://github.com/bitcointranscripts/registry/issues/53

**Review by kouloumos — COMMENTED** [review]
This looks good!
- The metadata for each source that [we talked about](https://github.com/bitcointranscripts/registry/pull/51#issuecomment-2443323832), are working great.
- Both `/types` and `/sources`page look awesome.

A couple of issues that I noticed:

### Level-2 sources

**1) Titles**
We are using the `title` as coming from the `_index.md` file for level 1 sources, but not for level 2 sources, like for example Advancing Bitcoin
advancing-bitcoin-sources-issue

In this case, we should read the  [_index.md files](https://github.com/kouloumos/bitcointranscripts/blob/refine-taxonomies/advancing-bitcoin/2020/_index.md) in order for this to become:
advancing-bitcoin-source-fix

same applies for other sources, such as Bitcoin++

**2) Number of Transcripts**: Based on the designs, sources at this level need to also display the number of transcripts in the source.

### Transcripts in Source
When looking at  the transcripts in a source, there are 3 issues
image

1) the source title is not displayed correctly, see for example "Mit Bitcoin Expo" instead of "MIT Bitcoin Expo".

2) If there is no date, the date icon on the top right is still displayed.

3) the transcripts are not sorted by date

**Pull request merged**